pypeit/core/procimg.py

```diff
@@ -2,20 +2,417 @@
+        sci_list: list
+            List of  float ndarray images (each being an image stack with shape (nimgs, nspec, nspat))
+            which are to be combined with the  weights, inmask_stack, and possibly sigma clipping
+        var_list: list
+            List of  float ndarray variance images (each being an image stack with shape (nimgs, nspec, nspat))
+            which are to be combined with proper erorr propagation, i.e.
+            using the  weights**2, inmask_stack, and possibly sigma clipping
+        inmask_stack: ndarray, boolean, shape (nimgs, nspec, nspat)
+            Array of input masks for the images. True = Good, False=Bad
+        sigma_clip: bool, default = False
+            Combine with a mask by sigma clipping the image stack. Only valid if nimgs > 2
+        sigma_clip_stack: ndarray, float, shape (nimgs, nspec, nspat), default = None
+            The image stack to be used for the sigma clipping. For example if
+            if the list of images to be combined with the weights is [sciimg_stack, waveimg_stack, tilts_stack] you
+            would be sigma clipping with sciimg_stack, and would set sigma_clip_stack = sciimg_stack
+        sigrej: int or float, default = None
+            Rejection threshold for sigma clipping. Code defaults to determining this automatically based
+            on the numberr of images provided.
+        maxiters:
+            Maximum number of iterations for sigma clipping using astropy.stats.SigmaClip
+
+    Returns:
+        sci_list_out: list
+           The list of ndarray float combined images with shape (nspec, nspat)
+        var_list_out: list
+           The list of ndarray propagated variance images with shape (nspec, nspat)
+        outmask: bool ndarray, shape (nspec, nspat)
+           Mask for combined image. True=Good, False=Bad
+        nused: int ndarray, shape (nspec, nspat)
+           Image of integers indicating the number of images that contributed to each pixel
+    """
+
+    shape = img_list_error_check(sci_list, var_list)
+    nimgs = shape[0]
+    nspec = shape[1]
+    nspat = shape[2]
+
+    if nimgs == 1:
+        # If only one image is passed in, simply return the input lists of images, but reshaped
+        # to be (nspec, nspat)
+        msgs.warn('Cannot combine a single image. Returning input images')
+        sci_list_out = []
+        for sci_stack in sci_list:
+            sci_list_out.append(sci_stack.reshape((nspec, nspat)))
+        var_list_out = []
+        for var_stack in var_list:
+            var_list_out.append(var_stack.reshape((nspec, nspat)))
+        outmask = inmask_stack.reshape((nspec, nspat))
+        nused = outmask.astype(int)
+        return sci_list_out, var_list_out, outmask, nused
+
+    if sigma_clip and nimgs >= 3:
+        if sigma_clip_stack is None:
+            msgs.error('You must specify sigma_clip_stack, i.e. which quantity to use for sigma clipping')
+        if sigrej is None:
```

Comment:
Can you comment on how this series of sigrej values
was decided upon?

Reply:
Not really, it was in some old code that I had. I think it was something from Schlegel's code somewhere. Anyway, I can make sigrej required if you prefer.